client/my-sites/domains/domain-management/components/domain/primary-flag.jsx

```diff
@@ -1,34 +1,27 @@
 /** @format */
-
 /**
  * External dependencies
  */
 
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
+import { flow } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import { getSelectedSiteId } from 'state/ui/selectors';
 import { isDomainOnlySite } from 'state/selectors';
 import { localize } from 'i18n-calypso';
-import Notice from 'components/notice';
 
 class DomainPrimaryFlag extends Component {
 	render() {
 		const { isDomainOnly, domain, translate } = this.props;
 
-		if ( domain.isPrimary && ! isDomainOnly ) {
-			return (
-				<Notice isCompact status="is-success">
-					{ translate( 'Primary Domain' ) }
-				</Notice>
-			);
-		}
-
-		return null;
+		return domain.isPrimary && ! isDomainOnly ? (
```

Comment:
Also, if there's no actual rule in the coding guidelines, let's use the old way, as it's way more readable than this. :) (again, subjective)

Reply:
What do you think to just making this a functional component? Would there be any downside?
Unless I'm mistaken, that would allow us to avoid the `null` side of the flow and just do:

```jsx
const DomainPrimaryFlag  =>  {
  if ( domain.isPrimary && ! isDomainOnly ) {
    return <span> { translate( 'Primary Domain' ) } </span>;
  }
}
```

Reply:
Totally ok with it. :)